### DISCUSSION TITLE
Error Command "npm run dev"

### DISCUSSION THREAD

**Discussion by ozxstudio** [tailwindlabs/tailwindcss#9091] [Help] [answered]
I'm about to try laravel with tailwind, having previously tried laravel with bootstrap. searching on google found the tailwindcss link with laravel. Unfortunately, when I run the npm run dev command, an error occurred, after I looked for the cause in "module.exports". in "tailwind.config.js" and in "postcss.config.js", please help resolve the error. thanks very much. best regards.

> ![tailwindlaravel](https://user-images.githubusercontent.com/67798242/184476267-7a2ac8a9-3668-4a9c-97ed-136f8efb00e4.png)

> ![tailwindlaravel2](https://user-images.githubusercontent.com/67798242/184476305-038de502-79dd-45d9-8955-c8bfd0b02fe9.png)

> ![tailwindlaravel3](https://user-images.githubusercontent.com/67798242/184476329-006aac2f-0a31-4eb4-9d1c-d6e4d85578ab.png)

**Comment by wongjn** [comment, marked as answer]
Could you try posting more of the error message? Or even the contents of the log file. It seems you only screenshot the last bit of the error which doesn't tell us much information to help troubleshoot.

**Reply by ozxstudio** [reply]
ok. thanks, I'll try to find

**Reply by ozxstudio** [reply]
could this be the error message?

0 verbose cli C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js
1 info using npm@8.11.0
2 info using node@v16.16.0
3 timing npm:load:whichnode Completed in 1ms
4 timing config:load:defaults Completed in 4ms
5 timing config:load:file:C:\Program Files\nodejs\node_modules\npm\npmrc Completed in 17ms
6 timing config:load:builtin Completed in 19ms
7 timing config:load:cli Completed in 4ms
8 timing config:load:env Completed in 1ms
9 timing config:load:file:D:\OZX STUDIO\laravel\laratail\.npmrc Completed in 0ms
10 timing config:load:project Completed in 4ms
11 timing config:load:file:C:\Users\user\.npmrc Completed in 0ms
12 timing config:load:user Completed in 1ms
13 timing config:load:file:C:\Users\user\AppData\Roaming\npm\etc\npmrc Completed in 1ms
14 timing config:load:global Completed in 1ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:credentials Completed in 3ms
17 timing config:load:setEnvs Completed in 2ms
18 timing config:load Completed in 40ms
19 timing npm:load:configload Completed in 42ms
20 timing npm:load:mkdirpcache Completed in 2ms
21 timing npm:load:mkdirplogs Completed in 1ms
22 verbose title npm run dev
23 verbose argv "run" "dev"
24 timing npm:load:setTitle Completed in 3ms
25 timing config:load:flatten Completed in 10ms
26 timing npm:load:display Completed in 18ms
27 verbose logfile logs-max:10 dir:C:\Users\user\AppData\Local\npm-cache\_logs
28 verbose logfile C:\Users\user\AppData\Local\npm-cache\_logs\2022-08-13T08_30_15_093Z-debug-0.log
29 timing npm:load:logFile Completed in 18ms
30 timing npm:load:timers Completed in 0ms
31 timing npm:load:configScope Completed in 0ms
32 timing npm:load Completed in 86ms
33 silly logfile start cleaning logs, removing 2 files
34 timing command:run Completed in 9ms
35 verbose stack Error: ENOENT: no such file or directory, open 'D:\OZX STUDIO\laravel\laratail\package.json'
36 verbose cwd D:\OZX STUDIO\laravel\laratail
37 verbose Windows_NT 10.0.19043
38 verbose node v16.16.0
39 verbose npm  v8.11.0
40 error code ENOENT
41 error syscall open
42 error path D:\OZX STUDIO\laravel\laratail/package.json
43 error errno -4058
44 error enoent ENOENT: no such file or directory, open 'D:\OZX STUDIO\laravel\laratail\package.json'
45 error enoent This is related to npm not being able to find a file.
45 error enoent
46 verbose exit -4058
47 timing npm Completed in 217ms
48 verbose code -4058
49 error A complete log of this run can be found in:
49 error     C:\Users\user\AppData\Local\npm-cache\_logs\2022-08-13T08_30_15_093Z-debug-0.log

**Reply by ozxstudio** [reply]
please help also to solve the problem with composer:

>![composer](https://user-images.githubusercontent.com/67798242/184477735-a823635a-30e1-469a-bf9b-2bd978a4a349.png)

**Reply by wongjn** [reply]
So the line of interest here is:

```
35 verbose stack Error: ENOENT: no such file or directory, open 'D:\OZX STUDIO\laravel\laratail\package.json'
```
So it seems like you haven't set up a `package.json` file. This file is what is needed to configure your project with respect to `npm` (and by extension, NodeJS).

**Reply by wongjn** [reply]
Not sure about the `composer` error – you may wish to ask somewhere else for that since this is a discussion area about Tailwind!

**Reply by ozxstudio** [reply]
OK, thanks a lot for solving the problem.

**Reply by ozxstudio** [reply]
yes, it seems that the problem with my windows.

**Reply by ozxstudio** [reply]
just a few more problems to run laravel tailwind, can't run the command "npm run dev"

sintax error:

0 verbose cli C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js
1 info using npm@8.11.0
2 info using node@v16.16.0
3 timing npm:load:whichnode Completed in 1ms
4 timing config:load:defaults Completed in 6ms
5 timing config:load:file:C:\Program Files\nodejs\node_modules\npm\npmrc Completed in 19ms
6 timing config:load:builtin Completed in 21ms
7 timing config:load:cli Completed in 3ms
8 timing config:load:env Completed in 2ms
9 timing config:load:file:D:\OZX STUDIO\laravel\laratail\.npmrc Completed in 0ms
10 timing config:load:project Completed in 3ms
11 timing config:load:file:C:\Users\user\.npmrc Completed in 1ms
12 timing config:load:user Completed in 2ms
13 timing config:load:file:C:\Users\user\AppData\Roaming\npm\etc\npmrc Completed in 0ms
14 timing config:load:global Completed in 1ms
15 timing config:load:validate Completed in 1ms
16 timing config:load:credentials Completed in 2ms
17 timing config:load:setEnvs Completed in 2ms
18 timing config:load Completed in 45ms
19 timing npm:load:configload Completed in 45ms
20 timing npm:load:mkdirpcache Completed in 2ms
21 timing npm:load:mkdirplogs Completed in 1ms
22 verbose title npm run dev
23 verbose argv "run" "dev"
24 timing npm:load:setTitle Completed in 2ms
25 timing config:load:flatten Completed in 10ms
26 timing npm:load:display Completed in 18ms
27 verbose logfile logs-max:10 dir:C:\Users\user\AppData\Local\npm-cache\_logs
28 verbose logfile C:\Users\user\AppData\Local\npm-cache\_logs\2022-08-13T13_56_37_530Z-debug-0.log
29 timing npm:load:logFile Completed in 16ms
30 timing npm:load:timers Completed in 0ms
31 timing npm:load:configScope Completed in 0ms
32 timing npm:load Completed in 87ms
33 silly logfile start cleaning logs, removing 2 files
34 silly logfile done cleaning log files
35 timing command:run Completed in 19ms
36 verbose stack Error: Missing script: "dev"
36 verbose stack
36 verbose stack To see a list of scripts, run:
36 verbose stack   npm run
36 verbose stack     at RunScript.run (C:\Program Files\nodejs\node_modules\npm\lib\commands\run-script.js:97:13)
36 verbose stack     at async module.exports (C:\Program Files\nodejs\node_modules\npm\lib\cli.js:78:5)
37 verbose cwd D:\OZX STUDIO\laravel\laratail
38 verbose Windows_NT 10.0.19043
39 verbose node v16.16.0
40 verbose npm  v8.11.0
41 error Missing script: "dev"
41 error
41 error To see a list of scripts, run:
41 error   npm run
42 verbose exit 1
43 timing npm Completed in 153ms
44 verbose code 1
45 error A complete log of this run can be found in:
45 error     C:\Users\user\AppData\Local\npm-cache\_logs\2022-08-13T13_56_37_530Z-debug-0.log

**Reply by wongjn** [reply]
You're missing the the script you're trying to run, in this case, `dev`:
```
36 verbose stack Error: Missing script: "dev"
```
